### DISCUSSION TITLE
Multiple script commands

### DISCUSSION THREAD

**Discussion by pijusz** [astral-sh/rye#150] [Q&A]
I've been trying to do multiple commands that would could replace bash scripts.

Goal would be to f.e. replace whole [FastAPI scripts](https://github.com/tiangolo/fastapi/tree/master/scripts) folder to `pyproject.toml` with scripts.
F.e.:
```yaml
[tool.rye.scripts]
lint = "ruff src && black src --check && isort src --check-only"
```

Does not work both on Windows and WSL Debian

**Comment by cnpryer** [comment]
A quick glance makes me think we just need to be a little more creative with parsing into args. You could probably submit this as an issue.

**Reply by cnpryer** [reply]
IIUC it's expecting the first argument to be a program to execute. So if you threw this in something like `lint.sh` and did
```toml
[tool.rye.scripts]
lint = "sh ./lint.sh"
```
I think it'd run (replace `sh` with another shell).

Ideally I'd do something like
```toml
[tool.rye.scripts]
lint = [
    "ruff check src",
    "black src --check",
    "isort src --check-only"
]
```
So if that's not currently doable, and this is how scripts are intended to be utilized, I could look into getting this to work.

**Reply by cnpryer** [reply]
Also not sure if you know this but you can use `ruff` for import sorting as well.

```toml
[tool.rye.scripts]
lint = [
    "ruff check src --extend-select I",
    "black src --check",
]
```

**Reply by cnpryer** [reply]
fwiw I confirmed that this works on my mac
```toml
[tool.rye.scripts]
lint = [
    "sh",
    "-c",
    "ruff check src --extend-select I && black src --check",
]
```

**Reply by mitsuhiko** [reply]
I think it would be nice to support more complex setups there without resorting to platform specifics like `sh`. It would also be nice to be able to set environment variables.

**Reply by cnpryer** [reply]
Definitely. I'll look at it next.

**Reply by cnpryer** [reply]
Opened #152 to track this.

**Reply by mitsuhiko** [reply]
Resolved in #153

**Reply by cnpryer** [reply]
⚡️

**Reply by cnpryer** [reply]
I think you meant #153

**Reply by mitsuhiko** [reply]
Indeed.
